### ISSUE TITLE
PD e2e test failing

### ISSUE THREAD

**Issue by vishh** [GoogleCloudPlatform/kubernetes#11758]
The following test has been flaky for a while now:
`Pod Disks should schedule a pod w/ a RW PD, remove it, then schedule it on another host`

cc @wojtek-t 

```shell
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/pd.go:123
Expected error:
    <*exec.ExitError | 0xc20802d380>: {
        ProcessState: {
            pid: 19959,
            status: 256,
            rusage: {
                Utime: {Sec: 0, Usec: 404000},
                Stime: {Sec: 0, Usec: 8000},
                Maxrss: 23704,
                Ixrss: 0,
                Idrss: 0,
                Isrss: 0,
                Minflt: 1067,
                Majflt: 0,
                Nswap: 0,
                Inblock: 0,
                Oublock: 0,
                Msgsnd: 0,
                Msgrcv: 0,
                Nsignals: 0,
                Nvcsw: 3126,
                Nivcsw: 611,
            },
        },
    }
    exit status 1
not to have occurred
```

**Comment by vishh** [comment]
cc @saad-ali

**Comment by wojtek-t** [comment]
This failed last 4 times on Jenkins e2e-gce suite

**Comment by davidopp** [comment]
Yeah this has failed 6 times in a row now -- every time since 5 AM today.

**Comment by mbforbes** [comment]
That one's been failing hard for GKE CI, and the other PD one has been flaking:

![sadness](https://cloud.githubusercontent.com/assets/1170062/8862753/ddf2c4ce-3147-11e5-9dd2-6e67f18730f5.png)

Any leads?

**Comment by saad-ali** [comment]
I'll take a look.

**Comment by vishh** [comment]
Could this failure be related to #10986 ?

**Comment by saad-ali** [comment]
Yep, confirmed this was caused by #10986 and fixed by #11561. The test stopped failing in Jenkins since #11561 was merged.

(The new test `Pod Disks should schedule a pod w/ a RW PD shared between multiple containers, write to PD, delete pod, verify contents, and repeat in rapid succession` is failing but I'll track that separately).

The problem here was that the correct namespace was not being passed to kubectlExec (`default` was being passed in), so the pod couldn't be found.

From the logs:
```
STEP: submitting host0Pod to kubernetes
INFO: Waiting up to 5m0s for pod pd-test-4edcc490-3153-11e5-81e9-42010af01555            status to be running
INFO: Waiting for pod pd-test-4edcc490-3153-11e5-81e9-42010af01555            in namespace 'e2e-tests-pod-disks-cxcs2' status to be 'running'(found phase: "Pending", readiness: false) (3.083686ms elapsed)
INFO: Waiting for pod pd-test-4edcc490-3153-11e5-81e9-42010af01555            in namespace 'e2e-tests-pod-disks-cxcs2' status to be 'running'(found phase: "Pending", readiness: false) (5.007084614s elapsed)
INFO: Waiting for pod pd-test-4edcc490-3153-11e5-81e9-42010af01555            in namespace 'e2e-tests-pod-disks-cxcs2' status to be 'running'(found phase: "Pending", readiness: false) (10.011019197s elapsed)
INFO: Found pod 'pd-test-4edcc490-3153-11e5-81e9-42010af01555' on node 'e2e-gce-minion-2bdt'
STEP: writing a file in the container
INFO: Running '/jenkins-master-data/jobs/kubernetes-e2e-gce/workspace/kubernetes/platforms/linux/amd64/kubectl kubectl --server=https://146.148.53.205 --kubeconfig=/var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/.kube/config exec --namespace=default pd-test-4edcc490-3153-11e5-81e9-42010af01555 -- /bin/sh -c echo '8674665223082153551' > '/testpd/tracker''
INFO: error running kubectl exec to write file: exit status 1
stdout=
stderr=Error from server: pods "pd-test-4edcc490-3153-11e5-81e9-42010af01555" not found
)
```

**Comment by wojtek-t** [comment]
@saad-ali thanks for fixing that (and sorry for breaking)
